frontend/src/hooks/useBlockUpdateAndFocusWhenDataRefresh.tsx

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useRef, useState } from 'react'
+
+export function useBlockUpdateAndFocusWhenDataRefresh(savedData) {
```

Comment:
Just a semantic note:

In English, we use `on` rather than `when` when we're talking about an event point, and `while` when we're talking about the duration of an event. This is why you'll almost never cross a `when` in variable names from native-English speakers.

So I would rather call that `useBlockUpdateAndFocusOnDataRefresh`.

Reply:
Ok for the hook, but `When` is commonly used in tests and other places : https://en.wikipedia.org/wiki/Given-When-Then